slack_bolt/app/app.py

```diff
@@ -1249,7 +1252,7 @@ def _init_context(self, req: BoltRequest):
         req.context["token"] = self._token
         if self._token is not None:
             # This WebClient instance can be safely singleton
-            req.context["client"] = self._client
+            req.context["client"] = create_copy(self._client)
```

Comment:
if the singleton client does not work for next-gen functions, you can simplify this part. The extra overhead by this change should be ignorable.

```python
    def _init_context(self, req: BoltRequest):
        req.context["logger"] = get_bolt_app_logger(app_name=self.name, base_logger=self._base_logger)
        req.context["token"] = self._token
        client_per_request: WebClient = WebClient(
            token=self._token,  # this can be None
            base_url=self._client.base_url,
            timeout=self._client.timeout,
            ssl=self._client.ssl,
            proxy=self._client.proxy,
            headers=self._client.headers,
            team_id=req.context.team_id,
        )
        req.context["client"] = client_per_request
```